index.html

```diff
@@ -2456,70 +2216,48 @@ <h3 id="applying">
         <p>
           The <dfn>steps for processing the <code>related_applications</code>
           member</dfn> are given by the following algorithm. The algorithm
-          takes a manifest as an argument. This algorithm returns a list of
-          application objects <var>applications</var>, which can be empty.
+          takes a <a>WebAppManifest</a> <var>manifest</var> as an argument.
+          This algorithm returns an
+          Array&lt;<a>ExternalApplicationResource</a>&gt;.
         </p>
         <ol>
-          <li>Let <var>applications</var> be an empty list.
+          <li>Let <var>A</var> be <var>manifest["related_applications"]</var>.
           </li>
-          <li>Let <var>unprocessed applications</var> be the result of calling
-          the <a>[[\GetOwnProperty]]</a> internal method of <var>manifest</var>
-          with argument "<code>related_applications</code>".
+          <li>Let <var>A'</var> be a new Array object created as if by the
+          expression [].
           </li>
-          <li>If <var>unprocessed applications</var> is an <a>array</a>, then:
+          <li>For <var>V</var> in <var>A</var>:
             <ol>
-              <li>For each <var>potential application</var> in the array:
+              <li>if neither <var>V["src"]</var> nor <var>V["url"]</var> are
+              <code>null</code>:
                 <ol>
-                  <li>Let <var>platform</var> be the result of running the <a>
-                    steps for processing the <code>platform</code> member of an
-                    application</a> with <var>potential application</var>.
-                  </li>
-                  <li>If <var>platform</var> is <code>undefined</code>, move
-                  onto the next item if any are left.
-                  </li>
-                  <li>Let <var>id</var> be the result of running the <a>steps
-                  for processing the <code>id</code> member of an
-                  application</a> with <var>potential application</var>.
+                  <li>Let <var>V'</var> be a new object created as if by the
```

Comment:
Can this just destructively update V?

Just delete Steps 1--4, and use *V* instead of *V'*. Should be equivalent, and simpler.

Reply:
We could, I guess. @marcoscaceres  ?

Reply:
Agree... I'm not sure what "`'`" signifies there.

Reply:
Very common in math books :-) but let me change

Reply:
ok done

Reply:
Having an *X'* as the updated value for *X* is fine. The issue here is that you went to all this trouble to create a clone which isn't necessary if you just updated it.

Again, this looks good now and no further action required.